### PR TITLE
Add "No Recovery Hearts" option

### PR DESCRIPTION
**Pull request by ThreePendants**
Added an option that would remove collectible hearts from drop tables to mimic the Hero Mode of later Zeldas.

### CODE REVIEW

**Comment by DobbyDigital** [comment]
Hi!

Where is this option set? I see the variable to toggle the collectible hearts, but I don't see anywhere where that variable is set (and only one file changed, so no option on the GUI will appear for it). 

Also, when trying to run it I get a syntax error on line 91:

**File "C:\Users\dobby\OneDrive\Desktop\GitHub repo\OoT Randomiser\OoT-Randomizer\Patches.py", line 91
    rom.write_bytes(0xB5D814, [0x00, 0x00, 0x00, 0x00. 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00])
                                                       ^
SyntaxError: invalid syntax**

**Comment by DobbyDigital** [comment]
Aha, there's a period instead of a comma in between two of the entries of the list on line 91, that fixes the syntax error.

You can also replace the `if world.no_collectible_hearts != 'true':` check on line 81 by a simpler `if world.no_collectible_hearts:` if the no_collectible_hearts attribute is set as a boolean instead of as a string.

**Comment by ThreePendants** [comment]
Hello Dobby, I'm sorry about my lack of a response. This is my first time actually using GitHub so I was unaware of... well basically anything. As you can see from the new code it was changed to a single byte that accomplishes much of what the original commit had done. I also added a proper toggle into the program.

File: data/settings_mapping.json [hunk 1]
```diff
@@ -305,6 +305,7 @@
           "text": "Misc",
           "row_span": [1,1,1],
           "settings": [
+			"no_collectible_hearts",
```
Comment (Zannick) [inline data/settings_mapping.json:308]: This should probably go below damage multiplier, as it's similar to that kind of challenge, and the first few options here are pretty commonly used.

Comment (ThreePendants) [reply data/settings_mapping.json:308]: I agree. That was only put there as a means of testing and I forgot to move them. Thank you for doing so (at least I think you did.)

**Comment by Zannick** [comment]
This was discussed elsewhere and someone suggested that this should just be named "Hero Mode" if we're going to draw the comparison to existing Hero Modes.

**Comment by ThreePendants** [comment]
I have no issues with that Zannick. It is the terminology used in Zelda games, after all, if people want to know more the tooltip will say exactly what it does.

File: data/settings_mapping.json (hunk 1 above)
Comment (Zannick) [reply data/settings_mapping.json:308]: I'll take care of that.

**Comment by Roman971** [comment]
I feel like this should include more than just dropped recovery hearts to be considered a "Hero Mode".
For instance, this doesn't prevent players from getting full health recovery from small, big or great fairies.

**Comment by Zannick** [comment]
Based on [Hero Mode](https://zelda.gamepedia.com/Hero_Mode), it's not common to remove fairies' health restoration in actual Hero modes, although some fairy drops were removed in one game. I'd rather leave them in anyway, to err on the approachable side.

Arguable whether this should also increase damage. =)

**Comment by ThreePendants** [comment]
Yes, as Zannick notes, Hero Mode in official Zelda doesn't often remove fairy drops. While it would be a very simple addition to change the healing of Fairies to less hearts or nothing, I believe I asked about doing this originally and the response was that it would screw with the logic of x4 damage and One Hit KO.

As for dealing double damage, well we already have a toggle for that, heh.

**Pull request merged**